uniffi_udl/src/converters/callables.rs

```diff
@@ -27,17 +27,12 @@ impl APIConverter<FieldMetadata> for weedle::argument::Argument<'_> {
 impl APIConverter<FieldMetadata> for weedle::argument::SingleArgument<'_> {
     fn convert(&self, ci: &mut InterfaceCollector) -> Result<FieldMetadata> {
         let type_ = ci.resolve_type_expression(&self.type_)?;
-        if let Type::Object { .. } = type_ {
-            bail!("Objects cannot currently be used in enum variant data");
-        }
```

Comment:
I love these PRs, we just need to delete the code that's stopping users from doing the things they want.